pkg/frontend/querymiddleware/limits_test.go

```diff
@@ -218,6 +218,66 @@ func TestLimitsMiddleware_MaxQueryLength(t *testing.T) {
 	}
 }
 
+func TestLimitsMiddleware_CreationGracePeriod(t *testing.T) {
+	now := time.Now()
+
+	tests := map[string]struct {
+		reqStartTime        time.Time
+		reqEndTime          time.Time
+		creationGracePeriod time.Duration
+		expectedEndTime     time.Time
+	}{
+		"should not manipulate time range if creation grace period is disabled": {
+			reqStartTime:        now.Add(-time.Hour),
+			reqEndTime:          now.Add(2 * time.Hour),
+			creationGracePeriod: 0,
+			expectedEndTime:     now.Add(2 * time.Hour),
+		},
+		"should not manipulate time range for a query in now + creation_grace_period": {
+			reqStartTime:        now.Add(-time.Hour),
+			reqEndTime:          now.Add(time.Hour),
```

Comment:
Given the `reqEndTime` matches with `time.Now() + creationGracePeriod` there's no guarantee the logic hasn't triggered. It's better if you change this test case to have `reqEndTime` to something like `now.Add(30 * time.Minute)`

Reply:
Fixed.